File: app/src/main/java/com/mecong/tenderalarm/alarm/DeviceBootReceiver.java
package com.mecong.tenderalarm.alarm;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class DeviceBootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i(AlarmUtils.TAG, "DeviceBootReceiver");
        if (Intent.ACTION_BOOT_COMPLETED.equalsIgnoreCase(intent.getAction())) {
            AlarmUtils.resetupAllAlarms(context);
        }
    }
}

File: app/src/main/java/com/mecong/tenderalarm/alarm/DeviceBootReceiver.kt
package com.mecong.tenderalarm.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.mecong.tenderalarm.alarm.AlarmUtils.resetupAllAlarms

class DeviceBootReceiver : BroadcastReceiver() {
  override fun onReceive(context: Context, intent: Intent) {
    //HyperLog.i(AlarmUtils.TAG, "DeviceBootReceiver")
    if (Intent.ACTION_BOOT_COMPLETED.equals(intent.action, ignoreCase = true)) {
      resetupAllAlarms(context)
    }
  }
}